src/streaming/net/HTTPLoader.js

```diff
@@ -195,30 +188,57 @@ function HTTPLoader(cfg) {
             }
         };
 
-        const _oncomplete = function() {
-            // Update request timing info
-            requestObject.startDate = requestStartTime;
-            requestObject.endDate = new Date();
-            requestObject.firstByteDate = requestObject.firstByteDate || requestStartTime;
-            httpResponse.resourceTiming.responseEnd = requestObject.endDate.getTime();
-
-            // If enabled the ResourceTimingApi we add the corresponding information to the request object.
-            // These values are more accurate and can be used by the ThroughputController later
-            _addResourceTimingValues(httpRequest, httpResponse);
-        }
+        /**
+         * Fired when a request has been aborted, for example because the program called XMLHttpRequest.abort().
+         */
+        const _onabort = function () {
+            _onRequestEnd(true)
+        };
 
         /**
-         * Fired when an XMLHttpRequest transaction completes.
-         * This includes status codes such as 404. We handle errors in the _onError function.
+         * Fired when progress is terminated due to preset time expiring.
+         * @param event
          */
-        const _onload = function () {
-            _oncomplete();
+        const _ontimeout = function (event) {
+            let timeoutMessage;
+            // We know how much we already downloaded by looking at the timeout event
+            if (event.lengthComputable) {
+                let percentageComplete = (event.loaded / event.total) * 100;
+                timeoutMessage = 'Request timeout: loaded: ' + event.loaded + ', out of: ' + event.total + ' : ' + percentageComplete.toFixed(3) + '% Completed';
+            } else {
+                timeoutMessage = 'Request timeout: non-computable download size';
+            }
+            logger.warn(timeoutMessage);
+
+            _onRequestEnd();
+        };
+
+        const _onRequestEnd = function (aborted = false) {
+            // Remove the request from our list of requests
+            if (httpRequests.indexOf(httpRequest) !== -1) {
```

Comment:
This will not have any effect when a request is aborted as `httpRequests` is set to `[]` anyways. But doesn't do any harm so we can keep as is.